lib/backport_session.js

```diff
@@ -226,4 +247,56 @@ export default class BackportSession extends Session {
       };
     });
   }
+
+  getCurrentBranch() {
+    return runSync('git',
+      ['rev-parse', '--abbrev-ref', 'HEAD']
+    ).trim();
+  }
+
+  updateUpstreamRefs(branchName) {
+    runSync('git',
+      ['fetch', this.upstream, branchName]
+    );
+  }
+
+  getBranchCommit(branch) {
+    return runSync('git',
+      ['rev-parse', branch]
+    ).trim();
+  }
+
+  isBranchUpToDateWithUpstream(branch) {
+    this.updateUpstreamRefs(branch);
+
+    const localCommit = this.getBranchCommit(branch);
+    const upstreamCommit = this.getBranchCommit(`${this.upstream}/${branch}`);
+
+    return localCommit === upstreamCommit;
+  };
+
+  isLocalBranchExists(branch) {
+    try {
+      // will exit with code 1 if branch does not exist
+      runSync('git',
+        ['rev-parse', '--verify', '--quiet', branch]
+      );
+      return true;
+    } catch (e) {
+      return false;
+    }
+  }
+
+  syncBranchWithUpstream(branch) {
+    const currentBranch = this.getCurrentBranch();
+
+    runSync('git',
+      [
+        currentBranch !== branch ? 'fetch' : 'pull',
+        this.upstream,
+        `${branch}:${branch}`,
+        '-f'
```

Comment:
Here's what I happen when I try the command locally:

```console
$ git fetch upstream main:main -f
fatal: refusing to fetch into branch 'refs/heads/main' checked out at '…/node'
```

Are you sure it does work?

Reply:
It's weird, it's in the docs of backporting and it's working for me

Are you currently on main branch